primitives/transaction-pool/src/pool.rs

```diff
@@ -217,11 +219,29 @@ pub trait TransactionPool: Send + Sync {
 	fn ready_transaction(&self, hash: &TxHash<Self>) -> Option<Arc<Self::InPoolTransaction>>;
 }
 
+/// Events that the transaction pool listens for.
+pub enum ChainEvent<TP: TransactionPool> {
+	/// New blocks have been added to the chain
+	Canonical {
```

Comment:
I think it should be just called `NewBlock`, as afair it gets triggered for every block (not only canonical ones). The doc above seems to confirm that as well.